lib/portlayer/exec/commit.go

```diff
@@ -56,27 +56,61 @@ func Commit(op trace.Operation, sess *session.Session, h *Handle, waitTime *int3
 
 		var res *types.TaskInfo
 		var err error
-		if sess.IsVC() && Config.VirtualApp.ResourcePool != nil {
-			// Create the vm
-			res, err = tasks.WaitForResult(op, func(op context.Context) (tasks.Task, error) {
-				return Config.VirtualApp.CreateChildVM(op, *h.Spec.Spec(), nil)
-			})
-		} else {
-			// Create the vm
-			res, err = tasks.WaitForResult(op, func(op context.Context) (tasks.Task, error) {
-				return sess.VMFolder.CreateVM(op, *h.Spec.Spec(), Config.ResourcePool, nil)
-			})
-		}
+
+		// Create the vm
+		res, err = tasks.WaitForResult(op, func(op context.Context) (tasks.Task, error) {
```

Comment:
This seems like cleanup unrelated to this feature, but I'm worried there's some subtle relationship I'm missing. If this change _is_ related to the feature, could you explain why it's needed?

Reply:
Clean up. Given I needed an `IsVIC` check I started with it in this test, but logic flow was just much more clear once cleaned.

Making this change _does_ mean that we **must** not be running with vApp configuration - iirc while deployment to one is no longer supported the upgrade logic didn't migrate from vApp to resource pool.